### PR TITLE
etcdmain: reject unreasonably high values of -election-timeout

### LINKED ISSUE #3139: etcd should reject unreasonably high values of heartbeat-interval and election-timeout
- [ ] modify the docs to reflect that etcd won't accept high values
- [ ] reject unreasonably high value

from https://github.com/coreos/etcd/issues/3118

/cc @Pensu

### PR DESCRIPTION
**Pull request by yichengq**
This helps users to detect setting problem early.

fixes #3139 

/cc @jedsmith Could you help to review this one?

### CODE REVIEW

File: etcdmain/config.go [hunk 1]
```diff
@@ -41,6 +41,17 @@ const (
 	clusterStateFlagExisting = "existing"
 
 	defaultName = "default"
+
+	// maxElectionMs specifies the maximum value of election timeout.
+	//
+	// A reasonable round-trip time for the continental united states is
+	// 130ms, and the time between US and japan is around 350-400ms.
+	// Because package gets delayed a lot, and network situation may be
```
Comment (crawford) [inline etcdmain/config.go]: packets get

File: etcdmain/config.go [hunk 2]
```diff
@@ -41,6 +41,17 @@ const (
 	clusterStateFlagExisting = "existing"
 
 	defaultName = "default"
+
+	// maxElectionMs specifies the maximum value of election timeout.
+	//
+	// A reasonable round-trip time for the continental united states is
```
Comment (xiang90) [inline etcdmain/config.go]: I think we should move this out of our code to our documentation. Our user should see it and understand it.

File: etcdmain/config.go [hunk 3]
```diff
@@ -293,6 +304,9 @@ func (cfg *config) Parse(arguments []string) error {
 	if 5*cfg.TickMs > cfg.ElectionMs {
 		return fmt.Errorf("-election-timeout[%vms] should be at least as 5 times as -heartbeat-interval[%vms]", cfg.ElectionMs, cfg.TickMs)
 	}
+	if cfg.ElectionMs > maxElectionMs {
+		return fmt.Errorf("-election-timeout[%vms] is unreasonably high, which is expected to be less than %vms", cfg.ElectionMs, maxElectionMs)
```
Comment (xiang90) [inline etcdmain/config.go]: -election-timeout[%vms] is too long. Set `-election-timeout` less than %vms. See ... for details.

**Comment by xiang90** [comment]
LGTM

**Comment by yichengq** [comment]
it is changed to:
```
$ ./bin/etcd -election-timeout 60000
2015/07/24 15:05:20 etcdmain: error verifying flags, -election-timeout[60000ms] is unreasonably long, and should be set less than 50000ms. See 'etcd --help'.
$ ./bin/etcd --help
...
	--election-timeout '1000'
		time (in milliseconds) for an election to timeout. See tuning documentation for details.
```

File: etcdmain/config.go [hunk 4]
```diff
@@ -293,6 +297,9 @@ func (cfg *config) Parse(arguments []string) error {
 	if 5*cfg.TickMs > cfg.ElectionMs {
 		return fmt.Errorf("-election-timeout[%vms] should be at least as 5 times as -heartbeat-interval[%vms]", cfg.ElectionMs, cfg.TickMs)
 	}
+	if cfg.ElectionMs > maxElectionMs {
+		return fmt.Errorf("-election-timeout[%vms] is unreasonably long, and should be set less than %vms", cfg.ElectionMs, maxElectionMs)
```
Comment (xiang90) [inline etcdmain/config.go]: i do not like the wording here again. `unreasonably` sounds too harsh. Can we simply change it to `too long`? Also this is too verbose.

Comment (yichengq) [inline etcdmain/config.go]: sure. changed.

**Comment by xiang90** [comment]
LGTM

**Comment by xiang90** [comment]
please squash the commits though

**Pull request merged**